src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -1061,4 +1061,21 @@ public void testModuleNameInMultiThreadMode() throws Exception {
         }
     }
 
+    /**
+     * This test is a workaround for the Jacoco limitations.
+     * A call to {@link System#exit(int)} will never return, so Jacoco coverage probe will be
+     * missing. Mocking the {@code System} class will turn {@code System.exit()} to noop.
+     *
+     * @throws Exception if error occurs
+     * @see <a href="https://github.com/jacoco/jacoco/issues/117">Jacoco issue 117</a>
+     * @noinspection CallToSystemExit
```

Comment:
As I told before, all suppression should be explained in javadoc comment. Please do.

Base on conversation with rnveach, bellow, just make a summary, some link to documentation might help to understand this syntax.

Reply:
Done. Actually, there is no reason to check the call to `System.exit`. In other tests, it is already tested with `ExpectedSystemExit`.